backend/modules/auth/provider.py

```
@@ -0,0 +1,454 @@
+        user_email = google_data["email"]
+        if force_create:
+            local_part, domain = user_email.split('@', 1)
+            user_email = f"{local_part}+google@{domain}"
+            counter = 1
+            while db.query(User).filter(User.email == user_email).first():
+                user_email = f"{local_part}+google{counter}@{domain}"
+                counter += 1
+        
+        try:
+            user_id = generate_unique_user_id(db, role)
+        except Exception as e:
```

Ruff (0.14.6)

Do not catch blind exception: `Exception`

(BLE001)